pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1293,6 +1349,10 @@ void recycle() {
             retryCount = 0;
             batchSizeByte = 0;
             numMessagesInBatch = 1;
+            if (chunkedMessageCtx != null) {
+                ReferenceCountUtil.safeRelease(chunkedMessageCtx);
+            }
```

Comment:
```suggestion
            ReferenceCountUtil.safeRelease(chunkedMessageCtx);
```

`safeRelease` can handle null value well. It calls `release` internally:

```java
    public static boolean release(Object msg) {
        // if `msg` is null, `msg instanceof ReferenceCounted` returns false
        return msg instanceof ReferenceCounted ? ((ReferenceCounted)msg).release() : false;
    }
```

So we usually use`safeRelease` just to simplify our code to avoid null checks and swallowing(printing) warnings for invalid reference count.